sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -193,10 +198,71 @@ private object CosmosRowConverter
         }
     }
 
+    private def getFullFidelityMetadata(objectNode: ObjectNode): Option[ObjectNode] = {
+      if (objectNode == null) {
+        None
+      } else {
+        val metadata = objectNode.get(FullFidelityChangeFeedMetadataPropertyName)
+        if (metadata != null && metadata.isObject) {
+          Some(metadata.asInstanceOf[ObjectNode])
+        } else {
+          None
+        }
+      }
+    }
+
+    private def parsePreviousImage(objectNode: ObjectNode): Any = {
+      val metadataNode = getFullFidelityMetadata(objectNode)
```

Comment:
What about using pattern?

getFullFidelityMetadata(objectNode) match

case Some(metadataNode) =>metadataNode.get(PreviousImagePropertyName).toString
case None => null

Reply:
Changed it to pattern matching - not perfect because of using many Java apis that could return null but two fewer lines and little easier to read. Line numbers could be reduced further with consequently applying fold/map/getOrElse/orNull but that becomes very hard to read - so I think this is a reasonable compromise